luaui/Include/blueprint_substitution/reports.lua

```diff
@@ -0,0 +1,182 @@
+-- luaui/Include/blueprint_substitution/reports.lua
+-- Contains reporting functions for api_blueprint.lua
+
+local M = {}
+
+-- Store references passed in during include
+local substitutionLogic = nil
+
+-- Local data for reports
+local sideTotals = {}
+local sideComplete = {}
+local uncategorizedUnits = {}
+
+local function escapeCsvField(value)
+    local str = tostring(value or "")
+    if string.find(str, '[,\"\n]') then
+        str = string.gsub(str, '"', '""')
+        return '"' .. str .. '"'
+    else
+        return str
+    end
+end
+
+function M.SetDependencies(subLogic)
+    Spring.Log("BlueprintReports", "info", "Receiving dependencies via SetDependencies...")
+    substitutionLogic = subLogic
+    if not substitutionLogic then
+        Spring.Log("BlueprintReports", "error", "Failed to receive necessary logic tables via SetDependencies!")
+    else
+        Spring.Log("BlueprintReports", "info", "Dependencies received successfully via SetDependencies.")
+    end
+end
+
+-- ===================================================================
+-- Reporting Functions
+-- ===================================================================
+
+function M.generateMappingReport()
+    Spring.Echo("\n--- Blueprint Mapping Report Start (CSV Format) ---")
+    if not substitutionLogic or not substitutionLogic.MasterBuildingData then 
+        Spring.Log("BlueprintReports", "error", "[Mapping Report] Substitution logic or MasterBuildingData not available or not initialized."); return 
+    end
+
+    local SIDES = substitutionLogic.SIDES
+    if not SIDES then
+         Spring.Log("BlueprintReports", "error", "[Mapping Report] SIDES not found in substitutionLogic."); return
+    end
+
+    local masterBuildingDataMinimal = substitutionLogic.MasterBuildingData
+
+    local reportHeader = table.concat({
+        "Category", "DescName", -- "NameSource", -- Removing NameSource for now
```

Comment:
If it's not there then just get rid of it.
```suggestion
        "Category", "DescName",
```